### PR TITLE
EVENT-726: Add Google Tag Manager (GTM)

### PR DESCRIPTION
**Pull request by OzzieOrca**
[Jira](https://jira.cru.org/browse/EVENT-726)

### CODE REVIEW

File: app/scripts/controllers/reviewRegistration.js [hunk 1]
```diff
@@ -164,10 +164,10 @@ angular
           return registration.completeRegistration(currentRegistration);
         })
         .then(function() {
-          //send analytics event
-          analytics.digitalData.eventID = conference.id;
-          analytics.digitalData.registeredEventName = conference.name;
-          analytics.track('registration');
+          analytics.track('registration', {
```
Comment (TheNoodleMoose) [inline app/scripts/controllers/reviewRegistration.js:167]: Although I think this is a better way to do it, I think Clark just wants us to send the event name ```registration``` and then access the digital Data layer from the GTM side like we were doing or Adobe.

File: app/index.ejs [hunk 2]
```diff
@@ -36,10 +35,49 @@
...
+        w[l].push({ 'gtm.start': new Date().getTime(), event: 'gtm.js' });
+        var f = d.getElementsByTagName(s)[0],
+          j = d.createElement(s),
+          dl = l != 'dataLayer' ? '&l=' + l : '';
+        j.async = true;
+        j.src =
+          'https://www.googletagmanager.com/gtm.js?id=' +
+          i +
+          dl +
+          '&gtm_auth=VMsIb-u97IZIyagKFgXe1w&gtm_preview=env-1&gtm_cookies_win=x';
```
Comment (wrandall22) [inline app/index.ejs:54]: I don't think we want `gtm_preview` in prod?

**Comment by TheNoodleMoose** [comment]
@OzzieOrca Talking with Brenton and Clark, they decided to not go with the history change to trigger page views, and instead use a custom event(virutal-page-view) to trigger one. We just need to add this event to the ```firePageViewEvent``` function, with the event name being ```virtual-page-view```.
Screen Shot 2020-11-18 at 9 31 08 AM

File: app/index.ejs (hunk 2 above)
Comment (OzzieOrca) [reply app/index.ejs:54]: I got that code from the Live environment at https://tagmanager.google.com/?authuser=1#/admin/accounts/1597949959/containers/35270741/environments?containerDraftId=2 but I'll use the one from the link on the main dashboard page.

File: app/scripts/controllers/reviewRegistration.js (hunk 1 above)
Comment (OzzieOrca) [reply app/scripts/controllers/reviewRegistration.js:167]: Would setting both ways hurt? Seems like it'd be better not to mutate global state.

Comment (TheNoodleMoose) [reply app/scripts/controllers/reviewRegistration.js:167]: Technically you can send the data and they can just ignore it on the gtm side, but still access the digitalData layer like before.

**Comment by OzzieOrca** [comment]
Was there a problem with history triggers or are they just used to the other way? Seems like less custom code in our apps would be better.

**Comment by TheNoodleMoose** [comment]
They just want to use it the other way since they know it will work. I guess Sarah Rose(our consultant from adswerve) said that the history change can be hit or miss so they don't want to give it a try. I agree, the less custom code the better.

**Comment by TheNoodleMoose** [comment]
Brenton mentioned maybe trying the history change after the migration is completed when we are cleaning up the adobe code.

**Review by TheNoodleMoose — APPROVED** [review]

**Pull request merged**